steward/src/cellars/cellar_v2.rs

```diff
@@ -0,0 +1,702 @@
+            };
+
+            Ok(CellarV2Calls::CallOnAdaptor(call).encode())
+        }
+        RemovePosition(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &RemovePositionCall::function_name(),
+                &cellar_id,
+            );
+            let call = RemovePositionCall {
+                index: params.index,
+                in_debt_array: params.in_debt_array,
+            };
+
+            Ok(CellarV2Calls::RemovePosition(call).encode())
+        }
+        SetHoldingPosition(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetHoldingPositionCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetHoldingPositionCall {
+                position_id: params.position_id,
+            };
+
+            Ok(CellarV2Calls::SetHoldingPosition(call).encode())
+        }
+        SetStrategistPayoutAddress(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetStrategistPayoutAddressCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetStrategistPayoutAddressCall {
+                payout: sp_call_parse_address(params.payout)?,
+            };
+
+            Ok(CellarV2Calls::SetStrategistPayoutAddress(call).encode())
+        }
+        SwapPositions(params) => {
+            log_cellar_call(CELLAR_NAME, &SwapPositionsCall::function_name(), &cellar_id);
+            let call = SwapPositionsCall {
+                index_1: params.index_1,
+                index_2: params.index_2,
+                in_debt_array: params.in_debt_array,
+            };
+
+            Ok(CellarV2Calls::SwapPositions(call).encode())
+        }
+        SetShareLockPeriod(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetShareLockPeriodCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetShareLockPeriodCall {
+                new_lock: string_to_u256(params.new_lock)?,
+            };
+
+            Ok(CellarV2Calls::SetShareLockPeriod(call).encode())
+        }
+
+        // This will ultimately need to be a governance function, but for Seven Sea's live testing we are keeping
+        // it here until they get a feel for what an appropriate value is.
+        SetRebalanceDeviation(params) => {
+            log_cellar_call(
+                CELLAR_NAME,
+                &SetRebalanceDeviationCall::function_name(),
+                &cellar_id,
+            );
+            let call = SetRebalanceDeviationCall {
+                new_deviation: string_to_u256(params.new_deviation)?,
+            };
+
+            Ok(CellarV2Calls::SetRebalanceDeviation(call).encode())
+        }
+    }
+}
+
+/// Encodes the swap params as ABI-encoded bytes to be passed as args to the underlying
+/// swap function
+fn encode_swap_params(params: SwapParams) -> Result<Vec<u8>, Error> {
```

Comment:
`SwapParams` and `OracleSwapParams` are defined as "common" protos, but we're defining their implementation in the `cellar_v2` file. I believe we duplication `SwapParams` encoding in `cellar_v1`. Should we move to a common location?